scripts/us_eia/opendata/process/coal.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""EIA Coal Dataset specific functions."""
+
+import common
+import logging
+import re
+
+
+def extract_place_statvar(series_id, counters):
+    """Given the series_id, extract the raw place and stat-var ID.
+
+    Args:
+        series_id: EIA series ID
+        counters: map for updating error statistics
+
+    Returns a (place, raw-stat-var, is_us_place) tuple.
+    """
+    # Pattern #1: COAL.{Measure}.{Region}-{Code}.{Period}
+    # Region could include 3-leter codes (e.g. MAT - Middle Atlantic)
+    m = re.match(r"^COAL\.([^._]+_?[^._]+)\.([A-Z]+)-([0-9]+)\.([AQM])$",
+                 series_id)
+    if m:
+        measure = m.group(1)
+        place = m.group(2)
+        code = m.group(3)
+        period = m.group(4)
+        sv_id = f'COAL.{measure}.{code}.{period}'
+        return (place, sv_id, True)
+
+    # Pattern #2: COAL.{EXPORT|IMPORT}_{Measure}.{Type}-{CountryIso}-{UsPortIso}.{Period}
+    # Pattern #3: COAL.{SHIPMENT}_{Submeasure}.{Source}-{Destination}-{Material}.{Period}
+    m = re.match(r"^COAL\.([A-Z]+)_([A-Z]+)\.([^-]+)-([^-]+)-([^.]+)\.([AQM])$",
+                 series_id)
+    if m:
+        activity = m.group(1)
+        measure = m.group(2)
+        if activity in ['EXPORT', 'IMPORT']:
+            # Pattern #2
+            # TODO: model destination / source port as well
+            type = m.group(3)
+            place = m.group(4)
+            port = m.group(5)
+            period = m.group(6)
+            sv_id = f'COAL.{activity}_{measure}.{type}.{period}'
```

Comment:
If we don't add port to the sv_id, could we end up with inconsistent series where for the same date and everything else the values are different ?